DataFormats/PortableTestObjects/src/classes.cc

```diff
@@ -0,0 +1,52 @@
+#define G__DICTIONARY
+
+#include "Rtypes.h"
+#include "RtypesImp.h"
+#include "TGenericClassInfo.h"
+#include "TVirtualObject.h"
+
+#include "classes.h"
+
+namespace ROOT {
+   // forward declaration
+   TGenericClassInfo *GenerateInitInstance(portabletest::TestHostCollection const*);
+}
+
+// read function for portabletest::TestHostCollection
+// called for every event
+static void setROOTReadStreamerFor_portabletest_TestHostCollection(char *target, TVirtualObject *oldObj)
```

Comment:
This needs a better name :-)

Would it be a good idea to make it a (`static` or regular) member function of `PortableHostCollection` ?

It could simplify the interface, but the main downside I can think of is that it would add a dependency on the ROOT internals to the CMS type.

Reply:
I'd avoid making `PortableHostCollection` dependent on ROOT.